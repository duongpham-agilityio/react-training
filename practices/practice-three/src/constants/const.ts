export const LIMIT_QUANTITY: number = 5;
