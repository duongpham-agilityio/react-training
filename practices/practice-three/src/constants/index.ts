export * from './url';
export * from './message';
export * from './variable';
export * from './title';
export * from './regex';
export * from './time';
export * from './status';
