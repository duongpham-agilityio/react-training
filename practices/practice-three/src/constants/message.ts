interface IMessages {
  EMPTY: string;
  FAIL_TO_FETCH: string;
  ADD_TO_CART_SUCCESS: string;
  ADD_TO_CART_FAIL: string;
  REMOVE_FORM_CART_SUCCESS: string;
  REMOVE_FORM_CART_FAIL: string;
}

export const MESSAGES: Readonly<IMessages> = {
  EMPTY: 'No data',
  FAIL_TO_FETCH: 'Something went wrong!! Please reload page.',
  ADD_TO_CART_SUCCESS: 'Add to cart successful!',
  ADD_TO_CART_FAIL: 'Add to cart fail!',
  REMOVE_FORM_CART_SUCCESS: 'Remove from cart successful!',
  REMOVE_FORM_CART_FAIL: 'Remove from cart fail!',
};
