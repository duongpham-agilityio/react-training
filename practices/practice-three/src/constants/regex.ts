export const REGEXPS = {
  EMAIL: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
};
