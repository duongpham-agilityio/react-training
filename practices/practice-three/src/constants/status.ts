export const STATUS = {
  NOT_FOUND: 404,
  UNAUTHORIZED: 401,
  SUCCESS: 200,
};
