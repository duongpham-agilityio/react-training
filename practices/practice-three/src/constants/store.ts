export const STORES = {
  CART: 'cart',
  FAVORITE: 'favorites',
  AUTHENTICATION: 'auth',
};
