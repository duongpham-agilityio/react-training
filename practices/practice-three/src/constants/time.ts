export const TIMES = {
  DEBOUNCE: 500,
  TOAST: 1000,
};
