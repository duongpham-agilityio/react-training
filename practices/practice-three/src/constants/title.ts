interface ITitles {
  ADD: string;
  REMOVE: string;
  ERROR: string;
  SUCCESS: string;
}

export const TITLES: Readonly<ITitles> = {
  ADD: 'Add',
  REMOVE: 'Remove',
  ERROR: 'Error',
  SUCCESS: 'Success',
};
