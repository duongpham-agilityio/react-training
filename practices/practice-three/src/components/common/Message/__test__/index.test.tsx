import { render } from '@testing-library/react';

// Constants
import { MESSAGES } from '@/constants';

// Components
import { Message } from '..';

describe('CartOutline', () => {
  it('Match to snapshot', () => {
    const { container } = render(<Message message={MESSAGES.EMPTY} />);

    expect(container).toMatchSnapshot();
  });
});
