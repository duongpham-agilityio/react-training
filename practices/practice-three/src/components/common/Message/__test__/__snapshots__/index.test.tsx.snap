// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartOutline Match to snapshot 1`] = `
<div>
  <div
    class="css-gmuwbf"
  >
    <p
      class="chakra-text css-1hmppls"
    >
      No data
    </p>
  </div>
</div>
`;
